src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+
+    public void BarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            IconBarConfig config = new IconBarConfig(bar);
+            config.Disabled = !config.Disabled;
+            new SegmentedBarConfig(this).handleBarDisabling(bar);
+        }
+    }
+}
+
+/// <summary>
+///   Configuration class for SegmentedBar class
+/// </summary>
+public class SegmentedBarConfig
+{
+    private SegmentedBar target;
+
+    public SegmentedBarConfig(SegmentedBar target)
+    {
+        this.target = target;
+    }
+
+    public float[] Size
+    {
+        get { return new float[]{ target.RectSize.x, target.RectSize.y }; }
+        set
+        {
+            target.RectSize = new Vector2(value[0], value[1]);
+            target.RectMinSize = target.RectSize;
+        }
+    }
+
+    public string Type
+    {
+        get { return target.Type; }
+        set { target.Type = value; }
+    }
+
+    public void updateAndMoveBars(Dictionary<string, float> data)
+    {
+        removeUnusedBars(target, data);
+
+        int location = 0;
+        foreach (var process in data)
```

Comment:
So you have made this a general method for filling in the values and the types for bars, but you still use the specific terms like `process` here? You should either name this non-generally or replace the terms used in the body with also general terms.